addons/registry/enable

```diff
@@ -1,48 +1,79 @@
-#!/usr/bin/env bash
-
-set -e
-
-source $SNAP/actions/common/utils.sh
-CURRENT_DIR=$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)
-source $CURRENT_DIR/../common/utils.sh
-
-regex_args='([a-zA-Z]+=[a-zA-Z0-9]+(,|))'
-
-read -ra ARGUMENTS <<<"$1"
-if [ -z "${ARGUMENTS[@]}" ]; then
-  echo "The registry will be created with the default size of 20Gi."
-  echo "You can use the \"size\" argument while enabling the registry, eg microk8s.enable registry:size=30Gi"
-  declare -A map
-  map[\$DISKSIZE]="20Gi"
-  "$SNAP/microk8s-enable.wrapper" hostpath-storage
-  echo "Applying registry manifest"
-  use_addon_manifest registry/registry apply "$(declare -p map)"
-  echo "The registry is enabled"
-elif [[ ${ARGUMENTS[@]} =~ $regex_args ]]; then
-	IFS=',' read -ra args <<< ${ARGUMENTS[@]}
-  REGEX_DISK_SIZE='(^[2-9][0-9]{1,}|^[1-9][0-9]{2,})(Gi$)'
-	for arg in "${args[@]}"; do
-		read -r key value <<<$(echo $arg | gawk -F "=" '{print $1 ,$2}')
-    if [ "$key" != "size" ]; then
-      echo "The only authorized argument is \"size\", eg  microk8s.enable registry:size=30Gi"
-      echo "The \"size\" value should match the regex: ([a-zA-Z]+=[a-zA-Z0-9]+(,|))"
-      exit 1
-    elif [ "$key" = "size"  ] && [[ ! $value =~ $REGEX_DISK_SIZE ]]; then
-      echo "The size of the registry should be higher or equal to 20Gi and match the regex: (^[2-9][0-9]{1,}|^[1-9][0-9]{2,})(Gi$)"
-      exit 1
-		elif [ "$key" = "size" ] && [[ $value =~ $REGEX_DISK_SIZE ]]; then
-      "$SNAP/microk8s-enable.wrapper" hostpath-storage
-      declare -A map
-      map[\$DISKSIZE]=$value
-    fi
-  done
-  echo "Enabling the private registry"
-  echo "Applying registry manifest"
-  use_addon_manifest registry/registry apply "$(declare -p map)"
-  echo "The registry is enabled"
-  echo "The size of the persistent volume is $value"
-else
-  echo "The only authorized argument is \"size\", eg  microk8s.enable registry:size=30Gi"
-  echo "The \"size\" value should match the regex: ([a-zA-Z]+=[a-zA-Z0-9]+(,|))"
-  exit 1
-fi
+#!/usr/bin/env python3
+import os
+import subprocess
+import shlex
+import re
+import yaml
+
+import click
+
+REGEX_DISK_SIZE = "(^[2-9][0-9]{1,}|^[1-9][0-9]{2,})(Gi$)"
+
+
+@click.command()
+@click.option("--size", default="20Gi", help="Size of the registry")
+@click.option("--storageclass", default=None, help="Storage class to be used")
+def enable_registry(size, storageclass):
```

Comment:
Note that this breaks backwards compatibility: https://microk8s.io/docs/registry-built-in.

It would be nice to support an optional `size=$size` argument, (and maybe print a warning telling people to use the new `--size` argument instead).

As an example, see how the `driver_type` and `driver` arguments are handled in https://github.com/canonical/microk8s-core-addons/blob/main/addons/gpu/enable#L28-L71